GSA-TTS/FAC
# 4434 implementing audit deletion workflow

## Description
This PR introduces functionality allowing users to remove unwanted reports from their submission list (refer to the [ARD](https://github.com/GSA-TTS/FAC/blob/main/docs/architecture/decisions/0041-in-progress-audit-deletion-workflow.md) for more context).
Only reports marked as "in progress" can be removed.
Users will see a delete icon next to any report in progress, which they can click to flag the report for removal.
Once a report is flagged for removal, it will be filtered out from the user's list of in-progress reports and will no longer be accessible to them.
Staff members can use the Django admin interface to revert a flagged report back to "in progress," restoring user access to it.

## How to Test
**Switch to this PR branch locally:**
- Launch the application stack.
- Create a report if you don't already have one in progress.

**Test the feature:**
- Observe that a delete icon is now available next to reports in progress, allowing you to flag them for removal.

![image](https://github.com/user-attachments/assets/e0c696a5-78f7-4f4e-8a6e-7578283ccd7f)

- Test the user access management feature (no changes were made to its functionality). You will see a new link at the bottom of the User Access Management UI to return directly to your list of submissions.
![image](https://github.com/user-attachments/assets/758c9ca6-6912-43cb-8bd0-7f175b91d70b)

**Test the remove report feature:**
- Click the delete icon to flag a report for removal.
- A confirmation screen will appear, allowing you to either confirm the removal or return to the submission list.
- Verify that each action (confirmation or return) works as expected.
- Once flagged for removal, the report will no longer appear in your submission queue.

**Verify report restoration via Django Admin:**
- Log in to Django Admin and navigate to the SF-SACs section.
- Search for your report and select it using the checkbox on the left.
- Open the actions dropdown, Try available actions, and confirm that the report can be successfully reverted back to "in progress."
![image](https://github.com/user-attachments/assets/3004f935-c7a1-4d4d-bcb3-8aee0b03818d)

## PR Checklist: Submitter


## PR Checklist: Reviewer


The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger
